Source/init.cpp

```diff
@@ -97,7 +97,11 @@ HANDLE init_test_access(char *mpq_path, char *mpq_name, char *reg_loc, int dwPri
 
 	for (int i = 0; i < 2; i++) {
 		snprintf(mpq_path, MAX_PATH, "%s%s", Buffer[i], mpq_name);
+#ifndef SWITCH
 		if (SFileOpenArchive(mpq_path, dwPriority, MPQ_FLAG_READ_ONLY, &archive)) {
+#else
+		if (SFileOpenArchive(mpq_path, dwPriority, 0, &archive)) {

```

Comment:
We should figure out what is causing StormLib to fail to open files in read-only mode in Switch and Amiga.

Reply:
I think it is an OS limitation on Switch, nothing to do with Stormlib.